packages/ui/src/utils/hooks/useCloseByEscKey.ts

```diff
@@ -0,0 +1,32 @@
+import { useKey } from 'react-use'
+
+type CallbackFunction = (e?: KeyboardEvent) => void
+
+// `callback stack` to track callers of useCloseByEscKey
+const _cbStack: CallbackFunction[] = []
+
+function _removeFromStack(cb: CallbackFunction) {
+  const idx = _cbStack.indexOf(cb)
+  if (idx > -1) {
+    _cbStack.splice(idx, 1)
+  }
+}
+
+export const useCloseByEscKey = (cb: CallbackFunction) => {
+  // 'Esc' -> IE/Edge specific value
+  const predicate = (event: KeyboardEvent) => event.key === 'Escape' || event.key === 'Esc'
+  useKey(predicate, (nativeEvent: KeyboardEvent) => {
+    // check if the last callback on the stack is this cb.
+    const lastCb = _cbStack.length ? _cbStack[_cbStack.length - 1] : null
+    if (cb && lastCb === cb) {
```

Comment:
@utkukaratas What if `cb` changes?
Consider this case:
```tsx
const Component = () => {
  useCloseByEscKey(() => { /* do something */ })

  return ...
}
```
With every render of the `Component` we will create a new callback that will never be removed.

I suggest we create a `cbRef` with `useRef` and use `cbRef.current` inside of our own callback that we execute on kye press.
We should also add `useEffect` and remove old `cb` from the stack if it ever changes.
WDYT?